tester.tcl: do_execsql_test_on_specific_db behaves differently in limbo and sqlite, if spotted a constraint error

This bug can indicate that limbo does not return errors properly on a constraint error.
To reproduce, create the following tcl test:
```sql
do_execsql_test_on_specific_db {:memory:} uniq_constraint {
    CREATE TABLE test (id INTEGER unique);
    insert into test values (1);
    insert into test values (1);
    SELECT * FROM test;
} {"UNIQUE constraint failed: test.id (19)"}
```

Run it with **limbo**, it'll produce this error:
```
Test FAILED: CREATE TABLE test (id INTEGER unique);
    insert into test values (1);
    insert into test values (1);
    SELECT * FROM test;
returned 'Runtime error: UNIQUE constraint failed: test.id (19)
1'
expected 'UNIQUE constraint failed: test.id (19)'
```
Notice how limbo produces 2 rows: one with an error message, second with a '1'.

Run it with **sqlite**, it'll give a very different error:
```
Error: stepping, UNIQUE constraint failed: test.id (19)
    while executing
"exec {*}$command"
    (procedure "evaluate_sql" line 3)
    invoked from within
"evaluate_sql $sqlite_exec $db_name $sql"
    (procedure "run_test" line 2)
    invoked from within
"run_test $::sqlite_exec $db_name $combined_sql $combined_expected_output"
    (procedure "do_execsql_test_on_specific_db" line 5)
    invoked from within
"do_execsql_test_on_specific_db {:memory:} uniq_constraint {
    CREATE TABLE test (id INTEGER unique);
    insert into test values (1);
    insert int..."
    (file "./testing/insert.test" line 215)
    invoked from within
"source $testdir/insert.test"
    (file "./testing/all.test" line 11)
make: *** [test-compat] Error 1
```

--- a/testing/insert.test
+++ b/testing/insert.test
@@ -529,6 +529,13 @@ do_execsql_test_on_specific_db {:memory:} null-value-insert-null-type-column {
     SELECT * FROM test;
 } {1|}
 
+# https://github.com/tursodatabase/turso/issues/1710
+do_execsql_test_in_memory_error_content uniq_constraint {
+    CREATE TABLE test (id INTEGER unique);
+    insert into test values (1);
+    insert into test values (1);
+} {UNIQUE constraint failed: test.id (19)}
+
 do_execsql_test_in_memory_error_content insert-explicit-rowid-conflict {
     create table t (x);
     insert into t(rowid, x) values (1, 1);

